db/trie/merklepatriciatree/branchnode.go

```diff
@@ -17,24 +17,21 @@ import (
 const radix = 256
 
 type branchNode struct {
-	mpt    *merklePatriciaTree
-	hashes map[byte][]byte
-	isRoot bool
-	ser    []byte
+	cacheNode
+	children map[byte]node
+	isRoot   bool
```

Comment:
so saving the nodes instead of the hash
does this cache help speed up the trie operation, and how?